File: Selenium/Project/Activity4.py
# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManager
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Navigate to the URL
    driver.get("https://alchemy.hguy.co/jobs/")
    # Find the Second heading on the page using xpath and find the text
    heading = driver.find_element(By.XPATH, "//h2[text()='Quia quis non']").text
    wait = WebDriverWait(driver, 10)
    assert (heading == 'Quia quis non')
    print("Second heading matches", heading)
    driver.close()

File: Selenium/Project/Activity4.java
package groupProject;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import java.time.Duration;

public class Activity4 {

    public static void main(String[] args) {
        // Set up Firefox driver
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        // Open the page
        driver.get("https://alchemy.hguy.co/jobs/");
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(5));
        String secondHeading = driver.findElement(By.xpath("//h2[text()='Quia quis non']")).getText();
        Assert.assertEquals(secondHeading, "Quia quis non");
        System.out.println("The second heading matches: " + secondHeading);
        driver.close();
    }
}